provider/resource_playbook.go

```diff
@@ -0,0 +1,574 @@
+
+			"tags": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Required:    false,
+				Optional:    true,
+				Description: "List of tags of plays and tasks to run.",
+			},
+
+			"limit": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Required:    false,
+				Optional:    true,
+				Description: "List of hosts to exclude from the playbook execution.",
+			},
+
+			"check_mode": {
+				Type:     schema.TypeBool,
+				Required: false,
+				Optional: true,
+				Default:  false,
+				Description: "If 'true', playbook execution won't make any changes but " +
+					"only change predictions will be made.",
+			},
+
+			"diff_mode": {
+				Type:     schema.TypeBool,
+				Required: false,
+				Optional: true,
+				Default:  false,
+				Description: "" +
+					"If 'true', when changing (small) files and templates, differences in those files will be shown." +
+					"Recommended usage with 'check_mode'.",
+			},
+
+			// connection configs are handled with extra_vars
+			"force_handlers": {
+				Type:        schema.TypeBool,
+				Required:    false,
+				Optional:    true,
+				Default:     false,
+				Description: "If 'true', run handlers even if a task fails.",
+			},
+
+			// become configs are handled with extra_vars --> these are also connection configs
+			"extra_vars": {
+				Type:        schema.TypeMap,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Required:    false,
+				Optional:    true,
+				Description: "A map of additional variables as: { key-1 = value-1, key-2 = value-2, ... }.",
+			},
+
+			"var_files": { // adds @ at the beginning of filename
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Required:    false,
+				Optional:    true,
+				Description: "List of variable files.",
+			},
+
+			// Ansible Vault
+			"vault_files": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Required:    false,
+				Optional:    true,
+				Description: "List of vault files.",
+			},
+
+			"vault_password_file": {
+				Type:        schema.TypeString,
+				Required:    false,
+				Optional:    true,
+				Default:     "",
+				Description: "Path to a vault password file.",
+			},
+
+			"vault_id": {
+				Type:        schema.TypeString,
+				Required:    false,
+				Optional:    true,
+				Default:     "",
+				Description: "ID of the desired vault(s).",
+			},
+
+			// computed
+			"play_first_time": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Used to check if the playbook is being played for the first time (first 'terraform apply'.",
+			},
+			// debug output
+			"args": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Computed:    true,
+				Description: "Used to build arguments to run Ansible playbook with.",
+			},
+			// envs
+			"env_vars": {
```

Comment:
Unless I'm misunderstanding what this is doing, I'm not sure if this is needed. Any environment variables in the current shell will already be passed to Ansible.

Reply:
This is a computed parameter. You can pass environment variables to ansible playbook with ``terraform apply``. For instance: ``env EXAMPLE_ENV_VAR=example terraform apply``. This ``EXAMPLE_ENV_VAR`` will then further be passed to the call of ansible playbook command as an extra variable. And ``env_vars`` then contains all environment variables passed through the ``terraform apply`` command.

Reply:
But there's no extra logic needed to pass environment variables through terraform to ansible. If I remove all the code around passing environment variables and use the following playbook:
```yaml
- hosts: all
  gather_facts: false
  vars:
    contents: "{{ lookup('env', 'FILE_CONTENTS') }}"
  tasks:
    - shell: echo {{ contents }} > /myfile
```
with the following command:
```
FILE_CONTENTS=foobar terraform apply
```
then it behaves as expected. The call to `exec.Command()` already passes the current environment to `ansible`. The code as written does more than just pass the environment through, it sets every environment variable in the current shell to an ansible variable.

Reply:
You're right. I've tried removing environment passing and it is as you've said. I've removed this parameter from the resource.